File: game_plugin_interface/src/sc/protocol/requests/StepRequest.java
package sc.protocol.requests;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

@XStreamAlias("step")
public class StepRequest implements ILobbyRequest
{
	@XStreamAsAttribute
	public String	roomId;

	public StepRequest(String roomId)
	{
		this.roomId = roomId;
	}
}

File: sdk/src/main/server-api/sc/protocol/requests/StepRequest.kt
package sc.protocol.requests

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/** Request by administrative client to send a MoveRequest to the current player.
 * Only works for paused games. */
@XStreamAlias("step")
data class StepRequest @JvmOverloads constructor(
        @XStreamAsAttribute
        var roomId: String,
        @XStreamAsAttribute
        var forced: Boolean = false
): AdminLobbyRequest
